src/poliastro/twobody/orbit.py

```diff
@@ -1497,3 +1504,294 @@ def plot(self, label=None, use_3d=False, interactive=False):
             from poliastro.plotting.core import OrbitPlotter2D
 
             return OrbitPlotter2D().plot(self, label=label)
+
+    @classmethod
+    def __groundtrack_orbit(
+        cls,
+        attractor,
+        Q,
+        i=None,
+        e=None,
+        prograde=True,
+        xtol=None,
+        xrtol=None,
+        tol=None,
+        rtol=None,
+        maxiter=None,
+    ):
+        # pq calculation
+        if attractor.solar_rotational_period == 0.0 * u.day:
+            raise AttributeError(
+                f"Attractor {attractor.name} has not solar rotational period implemented"
+            )
+        nsol = 1.059e-7 * (
+            u.rad / u.s
+        )  # CHECK HOW TO CALCULATE THIS AND AVOID USING CONSTANT
+        Psol = attractor.solar_rotational_period.to(u.s)
+        Psid = attractor.rotational_period.to(u.s)
+        if i is not None:
+            if i <= np.pi / 2 * u.rad:  # Prograde
+                Pq = Psol * Psid / (Psid * (Q - 1) + Psol)
+            else:  # Retrograde
+                Pq = Psol * Psid / (Psid * (Q + 1) - Psol)
+        elif prograde:
+            Pq = Psol * Psid / (Psid * (Q - 1) + Psol)
+        else:
+            Pq = Psol * Psid / (Psid * (Q + 1) - Psol)
+
+        orbit_elements = {"e": e, "i": i, "nsol": nsol, "attractor": attractor}
+        a = variable_step_bisection_algorithm(
+            f=calculate_pq,
+            x0=attractor.R.to(u.km),
+            y=Pq,
+            kwargs=orbit_elements,
+            xtol=xtol,
+            xrtol=xrtol,
+            tol=tol,
+            rtol=rtol,
+            maxiter=maxiter,
+        )
+        if e is None:
+            e = calculate_e_from_a(a, nsol, i, attractor)
+        if i is None:
+            i = calculate_i_from_a(a, nsol, e, attractor)
```

Comment:
I propose that we create a new module, `poliastro.twobody.repeating_groundtrack`, and we move there all these lines, as well as the new functions. This way:

- We avoid having too much logic in `Orbit`
- We avoid having to import the `calculate_*_from_*` and `variable_step_bisection_algorithm`, which are only used once

Reply:
I think it's a good idea to move all the functions required to a new module. I will get into it as soon as possible.